include/onnxruntime/core/session/provider_stubs.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/session/onnxruntime_c_api.h"
+
+/*
+Stubs for the function that adds the EP to the session options for EPs that are not supported in this build. 
+
+There are two ways this addition to SessionOptions currently happens:
+
+1) Via a function that is exported directly from the EP code (usually from the provider factory). 
+   The naming convention is OrtSessionOptionsAppendExecutionProvider_<EP>
+
+2) Via an entry in the OrtApis struct.
+   The naming convention is SessionOptionsAppendExecutionProvider_<EP>
+
+When an EP is an external library the provider bridge will provide the entry point.
+
+NOTE: The function of style #1 is easier to use from places like the C# and Java bindings as it does not use
```

Comment:
I do plan to add the struct support to Java for EP creation, but haven't found the time yet. It's a non-trivial amount of work but should be fairly straightforward.